src/printer.js

```diff
@@ -2053,6 +2053,20 @@ function printJSXElement(path, options, print) {
         } else {
           children.push(print(childPath));
         }
+
+        if (
+          child.closingElement &&
+            child.closingElement.type === 'JSXClosingElement'
+        ) {
+          const nextChild = n.children[index + 1];
+
+          if (
+            typeof nextChild === 'undefined' ||
+              nextChild.type !== 'JSXText'
```

Comment:
I think this breaks if there are spaces between the elements... Can you add a test case for that?